### ISSUE TITLE
Codon navigation service with hamburger menu

### ISSUE THREAD

**Issue by abrasat** [CodonFramework/Samples#6]
Is it possible to use the codon framework navigation service combined with the xamarin.forms hamburger menu ( [master-detail page](https://developer.xamarin.com/guides/xamarin-forms/application-fundamentals/navigation/master-detail-page/) ) ?
If yes, could you please add it to the codon framework  xamarin.forms sample, or post a code snippet here  ?

**Comment by abrasat** [comment]
Is it posssible to use the NavigationService from the sample to set the Detail property of the MasterDetailPage to a new page, when the selected item from the master menu has changed ?

**Comment by DVaughan** [comment]
Yes, you could register an Action with the RoutingService. But, I'd probably publish a message using the IMessenger. Subscribe to the message in your MasterDetail page's view-model; then have the view-model coordinate displaying the detail page.

**Comment by abrasat** [comment]
Thank you for your solution proposal. Should the message contain the Detail page to be diplayed, or where should the MasterDetail view model get that information from ? And the MasterDetail view model has to access the MasterDetail page UI object in view to set the Detail property.

**Comment by DVaughan** [comment]
It depends on your architecture. Ideally I prefer not to send UI elements via the messenger. I would expose a list of view-models for each detail page from your MasterDetail view-model. I would have a property like CurrentDetailViewModel. When that property changes, the MasterDetail page can switch to the detail page corresponding to the CurrentDetailViewModel.

**Comment by abrasat** [comment]
Ok, thanks. What solution would you recommend to handle in the view-model code the IsPresented flag used by the MasterDetail page (needs to be reset when a new Detail page is selected) ?

Regarding the Codon INavigationService interface, it could be also helpful to extend it in a future release, to be able to handle navigation also when using a MasterDetail page in the application (for instance a flag hat allows the reset of the NavigationStack, when the selected Detail item changes).

**Comment by abrasat** [comment]
... also a "ChangeDetailPage" method would be useful in INavigationService, when using MasterDetailPages.
